core-common/src/main/java/org/glassfish/jersey/message/internal/HeaderUtils.java

```diff
@@ -303,12 +303,12 @@ public static void checkHeaderChanges(final Map<String, String> headersSnapshot,
     /**
      * Compare two NewCookies having the same name. See documentation RFC.
      *
-     * @param first    NewCookie to be compared.
-     * @param second NewCookie to be compared.
+     * @param first     NewCookie to be compared.
+     * @param second    NewCookie to be compared.
      * @return the preferred NewCookie according to rules :
      *              - the latest maxAge.
-     *              - if equal, compare the expiry date
-     *              - if equal, compare name length
+     *              - if equal, compare the expiry date.
+     *              - if equal, compare path length.
      */
     public static NewCookie getPreferredCookie(NewCookie first, NewCookie second) {
```

Comment:
I would advise to rewrite the whole method into something like:
```
public static NewCookie getPreferredCookie(NewCookie first, NewCookie second) {

        return Comparator.nullsFirst(
                Comparator.comparingInt(NewCookie::getMaxAge)
                        .thenComparing(NewCookie::getExpiry, Comparator.nullsLast(Comparator.naturalOrder()))
                        .thenComparing(NewCookie::getPath, Comparator.nullsLast(Comparator.comparing(String::length))))
                        .compare(first, second) > 0 ? first : second;
    }
```
because it starts to remind a nightmare with all those conditions.

Reply:
Good idea !